cmake/HalideGeneratorHelpers.cmake

```diff
@@ -343,21 +343,27 @@ endfunction()
 
 function(_Halide_target_link_gpu_libs TARGET VISIBILITY)
     if ("${ARGN}" MATCHES "opengl")
-        if (NOT TARGET X11::X11)
+        if (NOT "${ARGN}" MATCHES "egl" AND NOT TARGET X11::X11)
             find_package(X11)
             if (NOT X11_FOUND)
                 message(AUTHOR_WARNING "X11 dependency not found on system.")
             endif ()
         endif ()
-        target_link_libraries(${TARGET} ${VISIBILITY} X11::X11)
+        if (X11_FOUND)
+            target_link_libraries(${TARGET} ${VISIBILITY} X11::X11)
+        endif ()
 
-        if (NOT TARGET OpenGL::GL)
+        if (NOT TARGET OpenGL::GL AND NOT TARGET OpenGL::EGL)
             find_package(OpenGL QUIET)
             if (NOT OPENGL_FOUND)
                 message(AUTHOR_WARNING "OpenGL dependency not found on system.")
             endif ()
         endif ()
-        target_link_libraries(${TARGET} ${VISIBILITY} OpenGL::GL)
+        if ("${ARGN}" MATCHES "egl")
+            target_link_libraries(${TARGET} ${VISIBILITY} OpenGL::EGL)
+        else ()
+            target_link_libraries(${TARGET} ${VISIBILITY} OpenGL::GL)
+        endif ()
```

Comment:
This needs some fixing too. No need to run the check because `OpenGL::EGL` is missing if the user didn't ask for EGL.

```cmake
if ("${ARGN}" MATCHES "egl")
    if (NOT TARGET OpenGL::EGL)
        find_package(OpenGL QUIET COMPONENTS EGL)
        if (NOT OPENGL_FOUND)
            message(AUTHOR_WARNING "OpenGL dependency not found on system.")
        endif ()
    endif ()
    target_link_libraries(${TARGET} ${VISIBILITY} OpenGL::EGL)
else ()
    if (NOT TARGET OpenGL::GL)
        find_package(OpenGL QUIET)
        if (NOT OPENGL_FOUND)
            message(AUTHOR_WARNING "OpenGL dependency not found on system.")
        endif ()
    endif ()
    target_link_libraries(${TARGET} ${VISIBILITY} OpenGL::GL)
endif ()
```